relay-statsd/src/lib.rs

```diff
@@ -82,6 +83,9 @@ pub struct MetricsClient {
     pub default_tags: BTreeMap<String, String>,
     /// Global sample rate.
     pub sample_rate: f32,
+    /// Capture mode. Only use in tests. Stores all the metric strings sent.
+    #[cfg(test)]
+    pub capture: Option<Arc<Mutex<Vec<String>>>>,
```

Comment:
Just realized we could get this whole think cheaper using a [SpyMetricsSink](https://docs.rs/cadence/latest/cadence/struct.SpyMetricSink.html):

> This is not a general purpose sink, rather it’s a sink meant for verifying metrics written during the course of integration tests.